app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java

```diff
@@ -41,17 +55,63 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         adapter = new KalendarAdapter(getActivity());
         DaggerApiComponent.builder().build().inject(this);
         floatingActionButton = view.findViewById(R.id.choosecalendar);
+
         recyclerView = view.findViewById(R.id.apilistcalendars);
         recyclerView.setAdapter(adapter);
+
         LinearLayoutManager recyclerLayoutManager = new LinearLayoutManager(this.getContext());
+
         recyclerView.setLayoutManager(recyclerLayoutManager);
+
         DividerItemDecoration dividerItemDecoration =
                 new DividerItemDecoration(recyclerView.getContext(),
                         recyclerLayoutManager.getOrientation());
+
         recyclerView.addItemDecoration(dividerItemDecoration);
         kalPref = new KalPref(this.getContext());
         getKalendarResponse(kalPref.clientToken());
 
+        recyclerView.addOnItemTouchListener(new RecyclerTouchListener(getActivity(), recyclerView, new ClickListener() {
+            @Override
+            public void onLongClick(View view, int position) {
+                AlertDialog.Builder alert = new AlertDialog.Builder(getActivity());
+                alert.setTitle("Hey mate!");
+                alert.setMessage("You sure wanna delete, yo?");
+                alert.setPositiveButton("OK", new DialogInterface.OnClickListener(){
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        String clientToken = kalPref.clientToken();
+
+                        List<GetKalendarResponse> kalendarList = adapter.getGetKalendarResponses();
+                        GetKalendarResponse kalendarToDelete = kalendarList.get(position);
+                        long deleteId = kalendarToDelete.getId();
+
+                        backendApi.deleteKalendar(clientToken, deleteId).enqueue(new Callback<PostKalendarResponse>() {
+                            @Override
+                            public void onResponse(Call<PostKalendarResponse> call, Response<PostKalendarResponse> response) {
+                                PostKalendarResponse postKalendarResponse = response.body();
```

Comment:
only remove the item from the view, if the request was succesful `adapter.removeAt(position);`

Reply:
and show a toast that it was succesful